### ISSUE TITLE
Zipcode is required

### ISSUE THREAD

**Issue by studio16** [mollie/PrestaShop#87]
### Expected behavior
zipcode empty -> go to payment screen

### Actual behavior
500 error

### Steps to reproduce the behavior
If u have pick-up in store items there is no zipcode required to fullfill the order. This results in a 500 error

### Environment
* PrestaShop version: 1.7.2.4 
* Mollie module version: 3.1.1.

**Comment by firstred** [comment]
I recommend to update the module. The postcode used to be mandatory in older versions.

**Comment by studio16** [comment]
i have 3.3.1 not 3.1.1, my bad, but zip is still required apparently

**Comment by firstred** [comment]
That is strange. It shouldn't be! We're going to investigate if some of the API requirements have changed and adjust the module if necessary.

**Comment by studio16** [comment]
i don't think it's the api cause the 500 error occurs on the module/mollie/payment?method=ideal&rand=
 prior to the redirect to mollie

**Comment by firstred** [comment]
Thanks for the info! It does contact the API right before redirecting, because it has to know the direct URL to the payment screen, so I think it makes sense that it crashes when contacting the API. The module itself seems to find all the required information, but omits the `postcode` field when it is empty to make the API aware that the postcode is not part of the address: https://github.com/mollie/PrestaShop/blob/1dd11cf2e54347bcb337ac3ade678aba0614351b/mollie.php#L2459-L2461

Here's an overview to help with debugging:

Once you click a payment method, you are being redirected to the `/module/mollie/payment` page, where the module creates a new payment via the API. Depending on the API you have chosen (Orders API or Payments API), you will see the postcode being added here:
https://github.com/mollie/PrestaShop/blob/1dd11cf2e54347bcb337ac3ade678aba0614351b/mollie.php#L2459-L2461
https://github.com/mollie/PrestaShop/blob/1dd11cf2e54347bcb337ac3ade678aba0614351b/mollie.php#L2471-L2473
or here for the Orders API:
https://github.com/mollie/PrestaShop/blob/1dd11cf2e54347bcb337ac3ade678aba0614351b/mollie.php#L2490-L2492
https://github.com/mollie/PrestaShop/blob/1dd11cf2e54347bcb337ac3ade678aba0614351b/mollie.php#L2505-L2507

My guess is that somehow the requirements in the API have changed and we might have to always fill the postcode field. Using an empty string `''` previously did not work, so maybe we can extend the code like this:
```php
if ($billing->postcode) {
    $paymentData['billingAddress']['postalCode'] = (string) $billing->postcode;
} else {
    $paymentData['billingAddress']['postalCode'] = '-';
}
```
You can try this out with a new build I just made: [mollie-v3.3.2_postcode_hotfix.zip](https://github.com/mollie/PrestaShop/files/2638743/mollie-v3.3.2_postcode_hotfix.zip)

We're going to ask internally if this change is required and keep you updated. Meanwhile you can try this approach to see if it lifts the postcode requirement.

**Comment by studio16** [comment]
ok trying it in a few seconds and i will keep u posted, thanks

**Comment by studio16** [comment]
yep this works

**Comment by firstred** [comment]
Great! Thank you for letting us know the results. We are going to run a few tests of our own and will update the module as soon as possible!

**Comment by firstred** [comment]
Marking as solved since v3.3.3. If you still need help with the module feel free to open a new issue!
